demos/fleet_provisioning/fleet_provisioning_with_csr/mqtt_operations.c

```diff
@@ -172,6 +172,18 @@
  * @brief The length of the MQTT metrics string.
  */
 #define METRICS_STRING_LENGTH                    ( ( uint16_t ) ( sizeof( METRICS_STRING ) - 1 ) )
+
+/**
+ * @brief The length of the outgoing publish records array used by the coreMQTT
+ * library to track QoS > 0 packet ACKS for outgoing publishes.
+ */
+#define OUTGOING_PUBLISH_RECORD_LEN              20
+
+/**
+ * @brief The length of the incoming publish records array used by the coreMQTT
+ * library to track QoS > 0 packet ACKS for incoming publishes.
+ */
+#define INCOMING_PUBLISH_RECORD_LEN              20
```

Comment:
Can we verify that we are not allocating more than we need? Asking because [here](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/376f78161706e8151a9ad72774efd08beb2d769c/demos/fleet_provisioning/fleet_provisioning_with_csr/core_mqtt_config.h#L67) the equivalent is only 10.